### DISCUSSION TITLE
Run as specific user in docker

### DISCUSSION THREAD

**Discussion by PierreRust** [jcorporation/myMPD#967] [Q&A]
Hi,

I'm playing with myMPD and it's a great mpd web client , the best I've seen so far, thanks for your work !

I'm running it as part of a docker compose stack, where all containers are run as a specific user (to make conf file management & backup easier).
When running myMPD, it automatically switches to the myMPD user, which is created in the [Dockerfile](https://github.com/jcorporation/myMPD/blob/master/contrib/packaging/docker/Dockerfile.alpine). However, that user uid and guid does not match the user I'm using everywhere else, which causes issues.

I've seen that the use of xUID, which is the standard way of dealing with that in docker to run as non-root inside the container,  was mentioned in the documentation at some point, and them [removed](https://github.com/jcorporation/myMPD/commit/4f0b27ec2b2c2d8b4c2c238ee8ef2bcc94af83ea). On the other hand, from what I see in the dockerfile history, these env. variables were never actually used.
Am I missing something of is there any reason for that situation ?

I'd love to be able to set the user and group id to be used inside the container. This would only require to change the dockerfile (and the documentation accordingly),  I can contribute a PR for that if you want.

**Comment by jcorporation** [comment]
A PR would be great. It should not modify the default behavior.

**Reply by PierreRust** [reply]
You've been fast, I did not have time to push the PR !
I've added some comments and an alternative approach in the (closed) issue #968
